### PR TITLE
Implemented Factory

### PR DESCRIPTION
**Pull request by wkronmiller**
Implemented the Unix Socket IPC factory, and associated classes.
Still need to serialize/deserialize message objects or simply switch to
using dictionaries.

### CODE REVIEW

File: endpoints.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import message_object
+
+class EndpointType(object):
+    """
```
Comment (djpetti) [inline endpoints.py:4]: Indentation

File: socket_ipc_connection.py [hunk 2]
```diff
@@ -4,7 +4,7 @@ class SocketIPCConnection(object):
     Socket connection object, returned by IPC factory clas. Subclasses will be
     specific to the connection target
     """
-    def __init__(self, socket):
+    def __init__(self, socket, message_object_class):
         """
         Construct a new 'SocketIPCConnection' object.
```
Comment (djpetti) [inline socket_ipc_connection.py:10]: How about you say what the args are.

**Comment by djpetti** [comment]
Sorry this is so late, I didn't see it for awhile.

**Comment by wkronmiller** [comment]
Will fix.

**Comment by wkronmiller** [comment]
Fixed listed errors.

**Pull request closed without merging**